### DISCUSSION TITLE
[Request] Commands Only One User Can Use.

### DISCUSSION THREAD

**Discussion by Jyojibh** [Sehelitar/Kick.bot#101] [Ideas]
In the Commands tab there is the ability to add either moderators etc but is there a way to implement the same thing for kick where it can pull users from kick to be able to only use those actions you assign to said user.

Ex:
Command - !Memes
User - SendMemesPlox

Action:
Trigger - !memes
Sub Action - Play Sound file from folder.

**Comment by Sehelitar** [comment]
There's no way to add users from an external source unfortunatly, so this is not possible yet.

**Comment by Sehelitar** [comment]
The only way to limit access to a command to a few specific users is to implement this in your action and check if the ``user`` argument matches someone in your list (with a bit of C# code maybe).
I might have something in mind for this.

**Comment by TheGeekno72** [comment]
> The only way to limit access to a command to a few specific users is to implement this in your action and check if the `user` argument matches someone in your list (with a bit of C# code maybe). I might have something in mind for this.

@Sehelitar @Jyojibh just add an "if user equals [ignore case] yourname do nothing then continue else do nothing then break"
if you want several users to be able to use it, duplicate the above as much as you want but replace the last part (break) with continue except for the second one, it'll check if the username of the command caller is in that if cascade and break action on the last if not
that ought to be enough for whitelisting users on a command triggered action
